Sources/SwiftInspectorAnalyzers/ImportsAnalyzer.swift

```diff
@@ -23,130 +23,13 @@
 // SOFTWARE.
 
 import Foundation
-import SwiftSyntax
+import SwiftInspectorVisitors
 
-public final class ImportsAnalyzer: Analyzer {
+extension StandardAnalyzer {
 
-  /// - Parameter cachedSyntaxTree: The cached syntax tree to return the AST tree from
-  public init(cachedSyntaxTree: CachedSyntaxTree = .init()) {
-    self.cachedSyntaxTree = cachedSyntaxTree
+  public func analyzeImports(fileURL: URL) throws -> [ImportStatement] {
```

Comment:
I was thinking more of using something like:

```swift
public struct StandardAnalyzer {
  init(cachedSyntaxTree: CachedSyntaxTree, visitor: SyntaxVisitor) { ... }
```

where then you'd have this simply be:

```swift
extension StandardAnalyzer {
  static func imports(cachedSyntaxTree: CachedSyntaxTree = .init()) -> StandardAnalyzer {
    StandardAnalyzer(cachedSyntaxTree: cachedSyntaxTree, visitor: ImportSyntaxReader())
  }
}
```

and keep the `public func analyze(fileURL: URL) throws -> Output` signature on the Analyzer.

This will require us a bit more refactoring since we'd need for each visitor to have a generic returning type (so we can know what `Output` is, so I'm fine doing this in the meantime and we can in the future migrate.

Reply:
Oooooooh. I like this idea. Let's circle back to this though since I'm not convinced we want a single output type on each visitor. Maybe we do... but I want to spend a little more time with the visitors before I commit to that.